Date: Dec 6, 2022
From: Aravindan NarasimhapuramChakravarthy <aravindan_narasimhapuramchakravarthy@comcast.com>
Source: COMCAST
Subject: Define macros that are used in the code.
libxkbcommon uses a .y parser file to recreate .c file.
The definition should be available in the parser file to take effect in the source .c file.
Signed-off-by: Aravindan NarasimhapuramChakravarthy <aravindan_narasimhapuramchakravarthy@comcast.com>
Index: libxkbcommon-0.5.0/src/xkbcomp/parser.y
===================================================================
--- libxkbcommon-0.5.0.orig/src/xkbcomp/parser.y
+++ libxkbcommon-0.5.0/src/xkbcomp/parser.y
@@ -31,6 +31,11 @@
  */
 
 %{
+/* Tokens */
+#define YYEMPTY -2
+#define YYerror 256
+#define YYUNDEF 257
+
 #include "xkbcomp-priv.h"
 #include "ast-build.h"
 #include "parser-priv.h"
